app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -135,6 +142,62 @@ class MovementsController @Inject() (
       case None    => updateMovementLargeMessage(movementId, triggerId)
     }
 
+  def updateMessage(movementId: MovementId, messageId: MessageId) =
+    Action.async(parse.json) {
+      implicit request =>
+        mapRequest(request.body)
+          .flatMap {
+            largeMessageMetadata => handleUpdateMessage(largeMessageMetadata, request, movementId, messageId)
+          }
+          .fold[Result](
+            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
+            x => x
+          )
+    }
+
+  private def handleUpdateMessage(largeMessageMetadata: LargeMessageMetadata, request: Request[JsValue], movementId: MovementId, messageId: MessageId)(implicit
+    hc: HeaderCarrier
+  ) =
+    largeMessageMetadata match {
+      case LargeMessageMetadata(Some(_), status) =>
+        (for {
+          messageType <- extract(request.headers).asPresentation
+          sourceFile  <- objectStoreService.getObjectStoreFile(largeMessageMetadata.objectStoreURI.get).asPresentation
```

Comment:
We don't need to extract the message data and get the file from object-store, so this can be just one for comprehension. Here all we care about is updating a message id which matches the message id in the request, held inside a movement whose _id matches the movement id sent in the request. We then update the url field with the object-store url if present, and update the status field with the status sent. We're not creating a new message, we're updating fields in an existing message.

Reply:
The metadata extraction would already have been done at this point

Reply:
updated